### PR TITLE
KIP-320: Allow fetchers to detect

### PR DESCRIPTION
**Pull request by emasab**
and handle log truncation. Python changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
-## vNext
+## v2.1.0
 
 - Added `set_sasl_credentials`. This new method (on the Producer, Consumer, and AdminClient) allows modifying the stored
   SASL PLAIN/SCRAM credentials that will be used for subsequent (new) connections to a broker (#1511).
 - Wheels for Linux / arm64 (#1496).
 - Added support for Default num_partitions in CreateTopics Admin API.
 - Added support for password protected private key in CachedSchemaRegistryClient.
 - Add reference support in Schema Registry client. (@RickTalken, #1304)
+- KIP-320: Add offset leader epoch methods to the TopicPartition and Messageclasses (#1540).
```
Comment (philipnee) [inline CHANGELOG.md:11]: just wondering if Messageclasses is a typo: is it MessageClasses?

File: src/confluent_kafka/src/confluent_kafka.c [hunk 2]
```diff
@@ -930,9 +980,10 @@ static PyObject *TopicPartition_str0 (TopicPartition *self) {
 
 	ret = cfl_PyUnistr(
 		_FromFormat("TopicPartition{topic=%s,partition=%"CFL_PRId32
-			    ",offset=%s,error=%s}",
+			    ",offset=%s,leader_epoch=%s,error=%s}",
```
Comment (pranavrth) [inline src/confluent_kafka/src/confluent_kafka.c:983]: Just a thought - Do we really need to print leader_epoch everytime? I think if it is none, we shouldn't print.

File: src/confluent_kafka/src/confluent_kafka.c [hunk 3]
```diff
@@ -913,15 +942,36 @@ static PyMemberDef TopicPartition_members[] = {
         { NULL }
 };
 
+static PyGetSetDef TopicPartition_getters_and_setters[] = {
```
Comment (pranavrth) [inline src/confluent_kafka/src/confluent_kafka.c:945]: Why is this needed? Are we trying to restrict the user from setting this field?

Adding this only for leader_epoch should affect other attributes.

File: src/confluent_kafka/src/Consumer.c
Comment (pranavrth) [inline src/confluent_kafka/src/Consumer.c:1]: Are all the functions covered?

**Comment by pranavrth** [comment]
Update the docs as well for Message and TopicPartition.
